src/pkg/services/m365/api/contacts_pager_test.go

```diff
@@ -0,0 +1,73 @@
+package api_test
+
+import (
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type ContactsPagerIntgSuite struct {
+	tester.Suite
+	cts clientTesterSetup
+}
+
+func TestContactsPagerIntgSuite(t *testing.T) {
+	suite.Run(t, &ContactsPagerIntgSuite{
+		Suite: tester.NewIntegrationSuite(
+			t,
+			[][]string{tester.M365AcctCredEnvs}),
+	})
+}
+
+func (suite *ContactsPagerIntgSuite) SetupSuite() {
+	suite.cts = newClientTesterSetup(suite.T())
+}
+
+func (suite *ContactsPagerIntgSuite) TestGetItemsInContainerByCollisionKey() {
+	t := suite.T()
+	ac := suite.cts.ac.Contacts()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	container, err := ac.GetContainerByID(ctx, suite.cts.userID, "contacts")
+	require.NoError(t, err, clues.ToCore(err))
+
+	conts, err := ac.Stable.
+		Client().
+		Users().
+		ByUserId(suite.cts.userID).
+		ContactFolders().
+		ByContactFolderId(ptr.Val(container.GetId())).
+		Contacts().
+		Get(ctx, nil)
+	require.NoError(t, err, clues.ToCore(err))
+
+	cs := conts.GetValue()
+	expect := make([]string, 0, len(cs))
+
+	for _, c := range cs {
+		expect = append(expect, api.ContactCollisionKey(c))
+	}
+
+	results, err := ac.GetItemsInContainerByCollisionKey(ctx, suite.cts.userID, "contacts")
+	require.NoError(t, err, clues.ToCore(err))
+	require.Less(t, 0, len(results), "requires at least one result")
+
+	for k, v := range results {
+		assert.NotEmpty(t, k, "all keys should be populated")
+		assert.NotEmpty(t, v, "all values should be populated")
+	}
+
+	for _, e := range expect {
+		_, ok := results[e]
+		assert.Truef(t, ok, "expected results to contain collision key: %s", e)
+	}
```

Comment:
I'm a bit split overall on whether we should be making a discrete folder with an item for this. Mostly I'm concerned about flakes when the purge script runs, though I don't recall the precise variables it looks at when purging contacts (or other things) and whether that could result in getting fewer items on the second call than the first. If we do use a dedicated folder, the below could probably be changed to `assert.ElementsMatch`
```suggestion
	assert.Subset(t, maps.Keys(results), expected, "missing collision keys")
```

Reply:
All test cases are pulling from the canonical root folders (inbox, calendars, contacts) none of which are affected by the purge.  We should be protected across changes here unless someone is manually tampering with those folders, in which case all bets are off for testing anyway.